database/seeders/GamesTableSeeder.php

```diff
@@ -42,23 +50,147 @@ public function run(): void
+            $num_achievements = $game->achievements()->published()->count();
+            $num_progression = random_int(3, max(3, (int) floor($num_achievements / 2)));
+            $num_win = 1;
+            if ($num_achievements > 7 && random_int(1, 30) === 1) {
+                $num_win++;
+
+                if ($num_achievements > 13 && random_int(1, 50) === 1) {
+                    $num_win++;
+                }
+            }
+            $num_remaining = $num_achievements - $num_progression - $num_win;
+            $num_missable = random_int(0, (int) floor($num_remaining / 3));
+            $num_remaining -= $num_missable;
+
+            $index = 1;
+            foreach ($game->achievements()->published()->get() as $achievement) {
+                $achievement->DisplayOrder = $index++;
+
+                $type = random_int(0, 2);
+                  while (true) {
+                    $valid = match ($type) {
+                        0 => ($num_remaining > 0),
+                        1 => ($num_missable > 0),
+                        default => true,
+                    };
+
+                    if ($valid) {
+                        break;
+                    }
+
+                    $type++;
+                }
+
+                if ($type === 1) {
+                    $achievement->type = AchievementType::Missable;
+                    $num_missable--;
+                } elseif ($type === 2) {
+                    if ($num_progression > 0) {
+                        $achievement->type = AchievementType::Progression;
+                        $num_progression--;
+                    } elseif ($num_win > 0) {
+                        $achievement->type = AchievementType::WinCondition;
+                        $num_win--;
+                    }
+                }
+                $achievement->save();
+            }
         });
 
-        Game::all()->each(function (Game $game) {
-            $leaderboardCount = random_int(0, 10);
+        Game::all()->each(function (Game $game) use ($developers) {
+            $user_id = $game->achievements()->first()->user_id ?? $developers[array_rand($developers)];
 
-            $leaderboards = Leaderboard::factory()->count($leaderboardCount)->make()->each(function (Leaderboard $leaderboard) use ($game) {
-                $leaderboard->GameID = $game->ID;
-                $leaderboard->Title = ucwords(fake()->words(2, true));
-                $leaderboard->Description = fake()->sentence();
-            });
+            $game->leaderboards()->saveMany(Leaderboard::factory()->count(random_int(0, 10))->create([
+                'GameID' => $game->ID,
+                'author_id' => $user_id,
+            ]));
+        });
 
-            $game->leaderboards()->saveMany($leaderboards);
+        $gameMetricsAction = new UpdateGameMetricsAction();
+        Game::all()->each(function (Game $game) use ($gameMetricsAction) {
+            $gameMetricsAction->execute($game);
```

Comment:
At this point, there won't be any unlocks, so this just calculates the number of achievements and total points.